### PR TITLE
feature BEMT Rotor Hover

### PR DESCRIPTION
**Pull request by francescogervasio**
BEMT is a method that executes the blade element momentum theory. It accepts as input some data of the
  helicopter's rotor, specifically:

  Altitude: the altitude at which the helicopter is hovering (m)
  V_infty: the helicopter horizontal fligth speed (m/s)
  MTOW: Maximum Take-Off Weight of the helicopter (kg);
  R_mr: main rotor radius (m);
  R_hub: rotor hub radius(m);
  c_mr: chord of the main rotor's blade (m);
  N_mr: number of blades;
  theta0: pitch angle at the root of the rotor (°);
  theta_tw: blade twist (tip minus root incidence) (°);
  Cla: lift curve slope of the adopted airfoil (1/rad);
  Cd0: profile drag coefficient of the adopted airfoil.

  It gives as output:

  Tc: Thrust coefficient;
  T: Thrust (N);
  Qc: Torque (Power) coefficient;
  Q: Torque (Nm);
  Tc_PR: Thrust coefficient (considering the Prandtl correction function);
  T_PR: Thrust (considering the Prandtl correction function) (N);
  Qc_PR: Torque (Power) coefficient (considering the Prandtl correction function);
  Q_PR: Torque (considering the Prandtl correction function) (Nm);
  P: Power  (W);
  P_PR: Power (considering the Prandtl correction function) (W);
  Pc_i: Induced Power coefficient;
  P_i: Induced Power (W);
  Pc_0: Parasite Power coefficient;
  P_0: Parasite Power (W);
  FM: Figure of Merit;
  FM_PR: Figure of Merit (considering the Prandtl correction function);
  Disk_Loading: Disk Loading T/A (N/m^2);
  Power_Loading: Power Loading T/P (N/W);
  lambda_i: Induced inflow ratio distribution along the main rotor radius;
  alpha: angle of attack distribution along the main rotor radius (rad);
  dTcdr_bar: Thrust coefficient distribution along the adimensionalized main rotor radius;
  dQcdr_bar: Torque (Power) coefficient distribution along the adimensionalized main rotor radius;
  dTcdr_bar_PR: Thrust coefficient distribution along the adimensionalized main rotor radius (considering the Prandtl
  correction function);
  dQcdr_bar_PR: Torque (Power) coefficient distribution along the adimensionalized main rotor radius (considering the
  Prandtl correction function);
  r_bar: dimensional stations along the main rotor radius (m).

### CODE REVIEW

File: BEMT_Hover.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    dPc_0 = np.zeros(len(r_bar))  # Incremental parasite Power coefficient of the rotor element
+
+
+## FOR CYCLE ALONG THE ROTOR
+
+    for j in range(len(r_bar)):
+
+     # The second degree equation for the induced inflow ratio has coefficient 1, C1 and C2
+
+        C1 = mu + Cla * sol / 8  # First degree coefficient
```
Comment (EttoreSaetta) [inline BEMT_Hover.py:82]: Move outside the loop

File: BEMT_Hover.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    dQcdr_bar = np.zeros(len(r_bar))  # Incremental Torque coefficient along the main rotor
+    dQcdr_bar_PR = np.zeros(len(r_bar))  # Incremental Torque coefficient along the main rotor
+                                         # considering the Prandtl correction function
+    dPc_i = np.zeros(len(r_bar))  # Incremental induced Power coefficient of the rotor element
+    dPc_0 = np.zeros(len(r_bar))  # Incremental parasite Power coefficient of the rotor element
+
+
+## FOR CYCLE ALONG THE ROTOR
+
+    for j in range(len(r_bar)):
```
Comment (EttoreSaetta) [inline BEMT_Hover.py:78]: You can try vectorizing this loop

File: BEMT_Hover.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        Cl[j] = Cla * alpha[j]  # Lift coefficient
+        Cd[j] = Cd0  # Drag coefficient
+
+
+        dTcdr_bar[j] = 0.5 * sol * Cl[j] * r_bar[j]**2  # Incremental Thrust coefficient along the main rotor
+
+        dTcdr_bar_PR[j] = dTcdr_bar[j] * F_prandtl[j]   # Incremental Thrust coefficient along the main rotor
+                                                        # considering the Prandtl correction function
+
+        print(dTcdr_bar_PR[j])
```
Comment (EttoreSaetta) [inline BEMT_Hover.py:104]: Remove print

File: BEMT_Hover.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    Qc = np.round(np.trapz(dQcdr_bar, r_bar),8)  # Torque (Power) coefficient
+    Qc_PR = np.round(np.trapz(dQcdr_bar_PR, r_bar),8)  # Torque (Power) coefficient considering the Prandtl
+                                                               # correction function
+
+    Pc_i = np.round(np.trapz(dPc_i, r_bar),8) # Induced Power (Torque) coefficient
+    Pc_0 = np.round(np.trapz(dPc_0, r_bar),8) # Parasite Power (Torque) coefficient
+
+    FM = ((Tc**(3/2) / np.sqrt(2)))/Qc # Figure of Merit
+    FM_PR = ((Tc_PR ** (3 / 2) / np.sqrt(2))) / Qc_PR  # Figure of Merit considering the Prandtl correction function
```
Comment (EttoreSaetta) [inline BEMT_Hover.py:132]: You can avoid computing `Tc`, `Qc` and `FM` with Prandtl's correction.
Just provide in output the uncorrected distributions and the `F_prandtl`.

**Review by EttoreSaetta — CHANGES_REQUESTED** [review]
Make these changes before merging.

File: BEMT_Hover.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import numpy as np
+
+def BEMT(Helicopter, V_infty):
+
+    """BEMT_Hover is a method that executes the Blade Element Momentum Theory with respect for an helicopter in hover or
```
Comment (rtogna) [inline BEMT_Hover.py:5]: Nel commento iniziale inserire:
dove trovare la teoria es.: R. Tognaccini,....., p. xyz
autore, data, versione

**Review by rtogna — CHANGES_REQUESTED** [review]

**Review by EttoreSaetta — APPROVED** [review]

**Pull request merged**